### PR TITLE
refactor(analytics): Adds tenant ID to Analytics Provider

### LINKED ISSUE #4521: [REFACTOR] Update AnalyticsPool to hold variables for tenant ID
Update the [AnalyticsProvider](https://github.com/juspay/hyperswitch/blob/be44447c09ea8814dc8b88aa971e08cc749db5f3/crates/analytics/src/lib.rs#L67-L72) (or [`AnalyticsConfig`](https://github.com/juspay/hyperswitch/blob/be44447c09ea8814dc8b88aa971e08cc749db5f3/crates/analytics/src/lib.rs#L593-L608)) to hold a variable for tenant-ID.

This can be achieved by simply adding a new field in the underlying structs for SqlxClient & ClickhouseClient.
Update the AnalyticsPool constructor method to accept a Tenant-ID.

Tenant-ID can be a new type field backed by a string;

pub struct TenantID(String);

For now the AnalyticsPool can be initialized with a fixed tenantID "default"

### PR DESCRIPTION
**Pull request by RaghavRox**
Closes #4521 
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Adds TenantID field to SqlxClient and ClickHouseClient

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Comment by RaghavRox** [comment]
I have added tenant_id param to from_conf method. Is that  the correct way? @lsampras

File: crates/analytics/src/lib.rs [hunk 1]
```diff
@@ -64,6 +64,14 @@ use self::{
     types::MetricsError,
 };
 
+#[derive(Clone, Debug)]
+pub struct TenantID(String);
```
Comment (lsampras) [inline crates/analytics/src/lib.rs:68]: Can you move this struct definition to the common_utils crate?

Comment (RaghavRox) [reply crates/analytics/src/lib.rs]: I moved it to types.rs in common_utils crate and added #[allow(unused)] in some places so clippy can stop complaining. Is that fine?

**Review by lsampras — APPROVED** [review]

**Comment by lsampras** [comment]
@RaghavRox the latest commit fails to compile due to some errors,
can you address the CI failures [here](https://github.com/juspay/hyperswitch/actions/runs/9034252368/job/24903420003?pr=4594#step:11:2309).

**Review by lsampras — APPROVED** [review]

**Review by ShivanshMathurJuspay — DISMISSED** [review]

**Review by lsampras — DISMISSED** [review]

**Review by ivor-juspay — DISMISSED** [review]

**Review by ShivanshMathurJuspay — APPROVED** [review]

**Review by ivor-juspay — DISMISSED** [review]

**Review by prajjwalkumar17 — DISMISSED** [review]

**Review by lsampras — APPROVED** [review]

**Comment by lsampras** [comment]
the changes here are covered in #4867

**Pull request closed without merging**